File: relisoft-hr/server/models/Asset.js
import mongoose from 'mongoose';

const assetSchema = new mongoose.Schema(
  {
    name: { type: String, required: true },
    type: {
      type: String,
      enum: ['laptop', 'monitor', 'keyboard', 'mouse', 'phone', 'tablet', 'other'],
      required: true,
    },
    brand: { type: String },
    model: { type: String },
    serialNumber: { type: String },
    assetTag: { type: String },
    purchaseDate: { type: Date },
    purchaseCost: { type: Number },
    warrantyExpiry: { type: Date },
    status: {
      type: String,
      enum: ['available', 'issued', 'lost', 'damaged', 'disposed'],
      default: 'available',
    },
    assignedTo: { type: mongoose.Schema.Types.ObjectId, ref: 'Employee' },
    issuedDate: { type: Date },
    returnDate: { type: Date },
    condition: {
      type: String,
      enum: ['new', 'good', 'fair', 'poor'],
      default: 'good',
    },
    notes: { type: String },
  },
  { timestamps: true }
);

export default mongoose.model('Asset', assetSchema);

File: relisoft-hr/server/Models/Asset.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace RelisoftHR.Models;

public class Asset : IHasRowVersion
{
    [Key]
    public int Id { get; set; }
    [Required, MaxLength(100)]
    public string Name { get; set; } = "";
    [MaxLength(50)]
    public string AssetTag { get; set; } = "";
    [MaxLength(50)]
    public string Category { get; set; } = ""; // Laptop, Monitor, Keyboard, Mouse, Phone, AccessCard, etc.
    [MaxLength(200)]
    public string? SerialNumber { get; set; }
    [MaxLength(50)]
    public string Status { get; set; } = "Available"; // Available, Assigned, Damaged, Retired
    public DateTime CreatedOn { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedOn { get; set; }
    public byte[]? RowVersion { get; set; }
}

public class EmployeeAsset
{
    [Key]
    public int Id { get; set; }
    public int EmployeeId { get; set; }
    public int AssetId { get; set; }
    public DateTime AssignedOn { get; set; } = DateTime.UtcNow;
    public DateTime? ReturnedOn { get; set; }
    [MaxLength(50)]
    public string Status { get; set; } = "Assigned"; // Assigned, Returned, Damaged

    [ForeignKey(nameof(EmployeeId))]
    public Employee? Employee { get; set; }
    [ForeignKey(nameof(AssetId))]
    public Asset? Asset { get; set; }
}
